module/x/peggy/client/cli/oracle_tx.go

```diff
@@ -0,0 +1,169 @@
+				EthereumChainID:       ethChainID,
+				BridgeContractAddress: types.NewEthereumAddress(ethContractAddress),
+				Orchestrator:          cosmosAddr,
+				Claims: []types.EthereumClaim{
+					types.EthereumBridgeDepositClaim{
+						Nonce:          nonce,
+						ERC20Token:     types.NewERC20Token(uint64(amount), tokenSymbol, tokenContractAddr),
+						EthereumSender: ethSenderAddr,
+						CosmosReceiver: receiverAddr,
+					},
+				},
+			}
+			if err := msg.ValidateBasic(); err != nil {
+				return err
+			}
+			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
+		},
+	}
+}
+
+func CmdSendETHWithdrawalRequest(cdc *codec.Codec) *cobra.Command {
+	return &cobra.Command{
+		Use:   "withdrawal [eth chain id] [eth contract address] [nonce]",
+		Short: "Submit an eth event observed by an orchestrator",
+		Args:  cobra.ExactArgs(3),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cliCtx := context.NewCLIContext().WithCodec(cdc)
+			inBuf := bufio.NewReader(cmd.InOrStdin())
+			txBldr := auth.NewTxBuilderFromCLI(inBuf).WithTxEncoder(utils.GetTxEncoder(cdc))
+			cosmosAddr := cliCtx.GetFromAddress()
+
+			ethChainID := args[0]
+			ethContractAddress := args[1]
+			nonce, err := parseNonce(args[2])
+			if err != nil {
+				return err
+			}
+			msg := types.MsgCreateEthereumClaims{
+				EthereumChainID:       ethChainID,
+				BridgeContractAddress: types.NewEthereumAddress(ethContractAddress),
+				Orchestrator:          cosmosAddr,
+				Claims: []types.EthereumClaim{
+					types.EthereumBridgeWithdrawalBatchClaim{
+						Nonce: nonce,
+					},
+				},
+			}
+			if err := msg.ValidateBasic(); err != nil {
+				return err
+			}
+			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
+		},
+	}
+}
+
+func CmdSendETHMultiSigRequest(cdc *codec.Codec) *cobra.Command {
+	return &cobra.Command{
+		Use:   "multisig-update [eth chain id] [eth contract address] [nonce]",
+		Short: "Submit an eth event observed by an orchestrator",
+		Args:  cobra.ExactArgs(3),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cliCtx := context.NewCLIContext().WithCodec(cdc)
+			inBuf := bufio.NewReader(cmd.InOrStdin())
+			txBldr := auth.NewTxBuilderFromCLI(inBuf).WithTxEncoder(utils.GetTxEncoder(cdc))
+			cosmosAddr := cliCtx.GetFromAddress()
+
+			ethChainID := args[0]
+			ethContractAddress := types.NewEthereumAddress(args[1])
+			nonce, err := parseNonce(args[2])
+			if err != nil {
+				return err
+			}
+			msg := types.MsgCreateEthereumClaims{
+				EthereumChainID:       ethChainID,
+				BridgeContractAddress: ethContractAddress,
+				Orchestrator:          cosmosAddr,
+				Claims: []types.EthereumClaim{
+					types.EthereumBridgeMultiSigUpdateClaim{
+						Nonce: nonce,
+					},
+				},
+			}
+			if err := msg.ValidateBasic(); err != nil {
+				return err
+			}
+			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
+		},
+	}
+}
+
+// todo: we mix nonces as int64 and base64 bytes at the moment
```

Comment:
why is it mixed?

Reply:
This is related to https://github.com/althea-net/peggy/issues/48 . For generalization I was moving towards a []byte representation. That would be more flexible. After MVP we should revisit the different nonce types.